tests/integration/olp-cpp-sdk-core/ApiLookupClientTest.cpp

```diff
@@ -593,4 +589,132 @@ TEST_F(ApiLookupClientTest, LookupApiAsync) {
   }
 }
 
+TEST_F(ApiLookupClientTest, CustomCatalogProvider) {
```

Comment:
An integration test does not mean copy/paste from unit tests and place it in another folder. It means you test **VersionedLayerClient** with the static URL and make sure that it works as you expect on the public API.

Reply:
Added VersionedLayerClient test.

Still, I would like to keep ApiLookupClient integration tests, because async `LookupApi` isn't used in dataservice::read.

Reply:
If it brings any benefit so be it, but there is not much difference here between unit and integration tests. But let me not be the party pooper, so please keep them.